src/FileConverter.ts

```diff
@@ -1,43 +1,67 @@
-} from "vscode";
+import * as vscode from "vscode";
 import { SarifVersion } from "./common/Interfaces";
 import { Utilities } from "./Utilities";
+import { ChildProcess, spawn } from "child_process";
 
 /**
  * Handles converting a non sarif static analysis file to a sarif file via the sarif-sdk multitool
  */
 export class FileConverter {
-    public static ConvertCommand = "extension.sarif.Convert";
+    private static extensionContext: vscode.ExtensionContext | undefined;
+
+    public static initialize(extensionContext: vscode.ExtensionContext): void {
+        FileConverter.extensionContext = extensionContext;
+        FileConverter.registerCommands(extensionContext);
+    }
+
+    private  static registerCommands(extensionContext: vscode.ExtensionContext): void {
+        extensionContext.subscriptions.push(
+            vscode.commands.registerCommand("extension.sarif.Convert", FileConverter.selectConverter),
+        );
+    }
 
     /**
      * Opens a quick pick list to select a tool, then opens a file picker to select the file and converts selected file
      */
-    public static selectConverter() {
-        let tool: string;
-        window.showQuickPick(Array.from(FileConverter.Tools.keys())).then((value: string) => {
-            if (value === undefined) {
-                return;
-            }
+    public static async selectConverter(): Promise<void> {
+        interface ToolQuickPickItem extends vscode.QuickPickItem {
+            readonly extensions: string[];
+        }
 
-            tool = value;
-            const dialogOptions = { canSelectFiles: true, canSelectMany: false, filters: {} } as OpenDialogOptions;
-            const toolExt = FileConverter.Tools.get(tool);
-            dialogOptions.filters[`${tool} log files`] = toolExt;
+        const quickPickItems: ToolQuickPickItem[] = [];
+        for (const [toolName, toolExtensions] of FileConverter.Tools.entries()) {
+            quickPickItems.push({
+                label: toolName,
+                extensions: toolExtensions
+            });
+        }
 
-            return window.showOpenDialog(dialogOptions);
-        }).then((uris: Uri[]) => {
-            if (uris !== undefined && uris.length > 0) {
-                FileConverter.convert(uris[0], tool);
-            }
+        const tool: ToolQuickPickItem | undefined = await vscode.window.showQuickPick(quickPickItems);
+        if (!tool) {
+            return;
+        }
+
+        const toolName: string = `${tool.label} log files`;
+        const filters: { [name: string]: string[] } = {};
+        filters[toolName] = tool.extensions;
+
+        const openUris: vscode.Uri[] | undefined = await vscode.window.showOpenDialog({
+            canSelectFiles: true,
+            canSelectMany: false,
+            filters
         });
+
+        if (!openUris) {
+            return;
+        }
+        if (openUris.length > 0) {
+            FileConverter.convert(openUris[0], tool.label);
```

Comment:
>label [](start = 52, length = 5)

Comment the fact that the tool name is the same as the tool-label and that it is passed straight through as an argument to the SDK converter.